src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -284,6 +312,50 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructors(
         return email;
     }
     
+    private List<EmailWrapper> generateFeedbackSessionClosedEmail(
+            CourseAttributes course, FeedbackSessionAttributes session,
+            List<InstructorAttributes> instructors, List<StudentAttributes> students) {
+
+        String template = EmailTemplates.USER_FEEDBACK_SESSION_CLOSED;
+        String subject = EmailType.FEEDBACK_CLOSED.getSubject();
+        List<EmailWrapper> emails = new ArrayList<EmailWrapper>();
+
+        for (InstructorAttributes instructor : instructors) {
+
+            String emailBody = Templates.populateTemplate(template,
+                    "${userName}", instructor.name,
+                    "${courseName}", course.getName(),
+                    "${courseId}", course.getId(),
+                    "${feedbackSessionName}", session.getFeedbackSessionName(),
+                    "${deadline}", TimeHelper.formatTime12H(session.getEndTime()),
+                    "${supportEmail}", Config.SUPPORT_EMAIL);
+
+            EmailWrapper email = getEmptyEmailAddressedToEmail(instructor.email);
+            email.setSubject(String.format(subject, course.getName(), session.getFeedbackSessionName()));
+            email.setContent(emailBody);
```

Comment:
See if you extract a smaller method to generate the email for one instructor/student. 
e.g.
```
for (...){
    emails.add(FeedbackSessionClosedEmailForInstructor(...));
}
```